src/Workspaces/Core/Portable/Workspace/ProjectSystem/FileWatchedPortableExecutableReferenceFactory.cs

```diff
@@ -138,23 +138,16 @@ public void StartWatchingReference(string fullFilePath)
     /// 0, the file watcher will be stopped. This is *not* safe to attempt to call multiple times for the same project
     /// and reference (e.g. in applying workspace updates)
     /// </summary>
-    public void StopWatchingReference(string fullFilePath, TReference? referenceToTrack, [CallerFilePath] string callerFilePath = "", [CallerLineNumber] int callerLineNumber = 0)
+    public void StopWatchingReference(string fullFilePath, TReference referenceToTrack, [CallerFilePath] string callerFilePath = "", [CallerLineNumber] int callerLineNumber = 0)
     {
         lock (_gate)
         {
-            var disposalLocation = callerFilePath + ", line " + callerLineNumber;
             if (!_referenceFileWatchingTokens.TryGetValue(fullFilePath, out var watchedFileReference))
             {
-                if (referenceToTrack != null)
-                {
-                    // We're attempting to stop watching a file that we never started watching. This is a bug.
-                    var existingDisposalStackTrace = _previousDisposalLocations.TryGetValue(referenceToTrack, out var previousDisposalLocation);
```

Comment:
This was a bit odd before in that the return value wasn't being used. Instead, this PR ensures that a referenceToTrack is passed in, and now uses the result of this TryGetValue call to determine if the reference has been seen before.

Reply:
Ah yes, digging through history I see this was subject to a few rewrites and that ended up to its current state.